base/linalg/eigen.jl

```diff
@@ -199,6 +198,9 @@ function eigvals!(A::StridedMatrix{<:BlasComplex}; permute::Bool=true, scale::Bo
     return LAPACK.geevx!(permute ? (scale ? 'B' : 'P') : (scale ? 'S' : 'N'), 'N', 'N', 'N', A)[2]
 end
 
+# promotion type to use for eigenvalues of a Matrix{T}
+eigtype(T) = (Base.@_pure_meta; promote_type(Float32, typeof(float(zero(T)/norm(one(T))))))
```

Comment:
I don't think the `Base.@_pure_meta` part is necessary here. At least it doesn't change anything for `Int` and `Rational{Int}`.

Reply:
Do you want the annotation to stay? These macro annotations are no particularly pretty so if they don't help, I think it would be better not to have it.